test/internet/test-dns-stackoverflow.js

```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const dns = require('dns');
+
+dns.resolve4('ticbrasil.com.br', function(err, addresses, family) {
+  assert.strictEqual(typeof addresses.length[addresses.length - 1],
+                     'undefined');
```

Comment:
Should this be this instead?

```suggestion
assert.strictEqual(typeof addresses.[addresses.length - 1], 'string');
assert.strictEqual(typeof addresses.[addresses.length], 'undefined');
```

Reply:
No, after the fix it will still return 1500 elements, with the 1200 last elements being undefined. It is a quickfix, c-ares have already fixed it in their repo and there will be a new release
The bugged version was returning 1500 elements with most of them being beyond the end of the stack

Reply:
Ah, OK, in that case it should be this, right?:

```suggestion
assert.strictEqual(typeof addresses[addresses.length - 1], undefined);
```

Reply:
Specifically, `addresses.length[addresses.length - 1]` should be `addresses[addresses.length - 1]`, right?

Reply:
lol, yes, of course
Should I change it or are you going for the cherry-pick? Is this PR still valid?